File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/TextInputTextAppearance.java
/* Copyright Airship and Contributors */

package com.urbanairship.android.layout.property;

import com.urbanairship.json.JsonException;
import com.urbanairship.json.JsonMap;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class TextInputTextAppearance extends TextAppearance {

    @Nullable
    private final Color hintColor;

    public TextInputTextAppearance(@NonNull TextAppearance textAppearance, @Nullable Color hintColor) {
        super(textAppearance);
        this.hintColor = hintColor;
    }

    @NonNull
    public static TextInputTextAppearance fromJson(@NonNull JsonMap json) throws JsonException {
        Color color = Color.fromJsonField(json, "place_holder_color");
        return new TextInputTextAppearance(TextAppearance.fromJson(json), color);
    }

    @Nullable
    public Color getHintColor() {
        return hintColor;
    }

}

File: urbanairship-layout/src/main/java/com/urbanairship/android/layout/property/TextInputTextAppearance.kt
/* Copyright Airship and Contributors */
package com.urbanairship.android.layout.property

import com.urbanairship.json.JsonException
import com.urbanairship.json.JsonValue

public class TextInputTextAppearance public constructor(
    textAppearance: TextAppearance,
    @JvmField public val hintColor: Color?
) : TextAppearance(textAppearance) {

    public companion object {
        private const val KEY_PLACEHOLDER_COLOR = "place_holder_color"

        @Throws(JsonException::class)
        public fun fromJson(json: JsonValue): TextInputTextAppearance {
            val content = json.requireMap()

            return TextInputTextAppearance(
                textAppearance = TextAppearance.fromJson(json),
                hintColor = content[KEY_PLACEHOLDER_COLOR]?.let(Color::fromJson)
            )
        }
    }
}
